### PR TITLE
Fix for trigger at_logon and at_system_start

### PR DESCRIPTION
**Pull request by Nimesh-Msys**
Previously, library was working to create only time-based triggers and not event based, like
trigger while a user(specific or any) log on a system and
trigger when system starts.
To handle the same, made a check while calling time based specific properties eg, 'RandomDelay' and provided a functionality to handle event based triggers eg, 'Delay'. 
Have also made some code improvements

- Trigger option `delay_duration` (in minutes) is added in event based triggers
- To create a trigger `at system start`, pass trigger_type: `TaskScheduler::AT_SYSTEMSTART`
- To create a trigger `at log on`, pass trigger_type: `TaskScheduler::AT_LOGON`
- For a specific user, pass user_id: "Domain\\Name" or "Administrator"
   For any user, user_id need not be used.

example: to create a task while log on for a specific user
```   trigger = {
        user_id: "Administrator",
        days: 5,
        days_interval: 5,
        days_of_week: 5,
        end_day: 10,
        end_month: 6,
        end_year: 2018,
        minutes_duration: 5,
        minutes_interval: 5,
        months: 5,
        delay_duration: 5,
        start_day: 5,
        start_hour: 5,
        start_minute: 5,
        start_month: 5,
        start_year: 2018,
        trigger_type: TaskScheduler::AT_LOGON,
        weeks: 5,
        weeks_interval: 5  }
```
Signed-off-by: Nimesh Patni <nimesh.patni@msystechnologies.com>

### CODE REVIEW

**Comment by Vasu1105** [comment]
@btm and @djberg96  please review and merge this PR

File: lib/win32/taskscheduler.rb [hunk 1]
```diff
@@ -39,10 +39,10 @@ class Error < StandardError; end
     TASK_EVENT_TRIGGER_ON_IDLE = 5
 
     # Trigger is set to run the task at system startup
-    TASK_EVENT_TRIGGER_AT_SYSTEMSTART = 6
+    TASK_EVENT_TRIGGER_AT_SYSTEMSTART = 7
 
     # Trigger is set to run the task when a user logs on
-    TASK_EVENT_TRIGGER_AT_LOGON = 7
+    TASK_EVENT_TRIGGER_AT_LOGON = 8
```
Comment (btm) [inline lib/win32/taskscheduler.rb]: Why are these changing? That doesn't match the documented values from Microsoft.

https://msdn.microsoft.com/en-us/library/windows/desktop/aa383620(v=vs.85).aspx

Comment (Nimesh-Msys) [reply lib/win32/taskscheduler.rb]: Hi @btm ,
While testing previous trigger value, `with user_id`, task for `log on for 'any user'` was created.
On further analysis, we found that its actual value should be 9  ( Ref: `Task Scheduler 2.0 Enumerated Types` : https://msdn.microsoft.com/en-us/library/windows/desktop/aa383915(v=vs.85).aspx )

As per the current library code, `it increments the constant by 1` and then creates required task !
Also, while displaying, first `index value is incremented by 1`, and then trigger's value is returned.
( Ref while creation: https://github.com/chef/win32-taskscheduler/blob/ole/lib/win32/taskscheduler.rb#L644:L663
 Ref while display:  https://github.com/chef/win32-taskscheduler/blob/ole/lib/win32/taskscheduler.rb#L800 )
Hence, we assigned `TASK_EVENT_TRIGGER_AT_LOGON = 8`

But I think we may use actual constants value without any incrementation in any of the methods. Could you please let us know your views about this

**Comment by btm** [comment]
@Vasu1105 @Nimesh-Msys this needs a rebase now. the others are all merged.

**Pull request merged**